crates/subspace-farmer/src/commitments.rs

```diff
@@ -114,8 +114,8 @@ impl Commitments {
         let mut commitment_databases = self.inner.commitment_databases.lock().await;
 
         // Check if database was already been removed
-        if let Some(db_entry) = commitment_databases.databases.get(&salt) {
-            if db_entry.db.lock().await.is_some() {
+        if let Some(db_entry) = commitment_databases.get_db_entry(&salt) {
```

Comment:
`get_db_entry` replaced `databases.get` in here.
But `get_db_entry` uses `peek` instead of `get`. This is intentional, right (like another bug fix)?

Reply:
It is a part of the abstraction. There is nothing externally of `CommitmentDatabases` that exposes LRU presence, so `db_entry` either exists or it doesn't, doesn't really matter why. And internally we do need to make it use `peek` indeed. I guess we should replace `LRU` with just a `VecDeque` in the future as we only have 2 entries and scanning trough them will be faster than using any kind of hashmap, especially when we will hit the target on the first try most of the time. As I said, a lot more refactoring needs to be done :slightly_smiling_face:

Reply:
Yes, I guess LRU is an overkill for only 2 entries :)